### DISCUSSION TITLE
File Override

### DISCUSSION THREAD

**Discussion by s-canchi** [nih-cfde/training-and-engagement#433] [Snakemake]
Can you override if you wanted to overwrite the exiting file? 
Related: Does snakemake only check if the file exists ?  Is there a way to check if it's a certain size, etc.. since file metadata can change.

**Comment by marisalim** [comment]
There are a few snakemake flag options that allow you to write over existing files (i.e., see this [FAQ answer](https://snakemake.readthedocs.io/en/stable/project_info/faq.html?highlight=force#how-do-i-force-snakemake-to-rerun-all-jobs-from-the-rule-i-just-edited)):

```
  --force, -f           Force the execution of the selected target or the first rule regardless of already
                        created output.
  --forceall, -F        Force the execution of the selected (or the first) rule and all rules it is dependent on
                        regardless of already created output.
  --forcerun [TARGET ...], -R [TARGET ...]
                        Force the re-execution or creation of the given rules or files. Use this option if you
                        changed a rule and want to have all its output in your workflow updated.
```
